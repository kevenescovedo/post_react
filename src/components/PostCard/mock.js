export const postCardPropsMock = {
    id: 1,
    title: 'tile1',
    body : 'body1',
    cover: 'https://site.com',

}