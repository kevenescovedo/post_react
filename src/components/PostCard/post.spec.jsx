import { render, screen } from "@testing-library/react"
import { debug } from "console";
import { PostCard } from ".";
import { postCardPropsMock } from "./mock";
// criar um mock falso de dados
const props = postCardPropsMock;
describe('<PostCard/>', () => {
it('should render PostCard Correctly', () => {
const {debug} = render(<PostCard {...props}/>);

//ver se tem imagemn o documento
expect(screen.getByRole("img", {name: props.title})).toBeInTheDocument();

//ve se ela tem o src e o src é igual o esperado
expect(screen.getByRole('img', {name: props.title})).toHaveAttribute('src', props.cover);
expect(screen.getByRole('heading', {name: props.title})).toBeInTheDocument();
//pegar o p, o a tag p não tem role
expect(screen.getByText(props.body)).toBeInTheDocument();
});
it('should match snapshot', () => {
    const {debug,container} = render(<PostCard {...props}/>);

    expect(container.firstChild).toMatchSnapshot();
    })
})

