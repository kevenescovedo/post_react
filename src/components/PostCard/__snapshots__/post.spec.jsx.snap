// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard/> should match snapshot 1`] = `
<div
  class="post"
>
  <img
    alt="tile1"
    src="https://site.com"
  />
  <div
    class="post-content"
  >
    <h1>
      tile1
    </h1>
    <p>
      body1
    </p>
  </div>
</div>
`;
